components/dropdown/dropdown-menu.js

```diff
@@ -0,0 +1,91 @@
+import { html, LitElement } from 'lit-element/lit-element.js';
+import { DropdownContentMixin } from './dropdown-content-mixin.js';
+import { dropdownContentStyles } from './dropdown-content-styles.js';
+
+class DropdownMenu extends DropdownContentMixin(LitElement) {
+
+	static get styles() {
+		return dropdownContentStyles;
+	}
+
+	constructor() {
+		super();
+		this.noAutoFocus = true;
+		this.noPadding = true;
+	}
+
+	firstUpdated(changedProperties) {
+		super.firstUpdated(changedProperties);
+
+		this.addEventListener('d2l-dropdown-open', this._onOpen);
+		this.addEventListener('d2l-dropdown-close', this._onClose);
+		this.addEventListener('d2l-menu-resize', this._onMenuResize);
+		this.addEventListener('d2l-menu-item-select', this._onSelect);
+		this.addEventListener('d2l-menu-item-change', this._onChange);
+	}
+
+	render() {
+		return html`
+			${this._renderContent()}
+			<div class="d2l-dropdown-content-pointer">
+				<div></div>
+			</div>
+		`;
+	}
+
+	_onMenuResize(e) {
+		this.__position(!this._initializingHeight, e.detail);
+		this._initializingHeight = false;
+	}
+
+	_onClose(e) {
+
+		if (e.target !== this) {
+			return;
+		}
+
+		// reset to root view
+		const menu = this.__getMenuElement();
+		menu.show({ preventFocus: true });
+	}
+
+	_onOpen(e) {
+
+		if (e.target !== this) {
+			return;
+		}
+		this._initializingHeight = true;
+
+		const menu = this.__getMenuElement();
+
+		menu.resize();
+
+		if (this.__applyFocus) {
+			setTimeout(() => {
+				menu.focus();
+			}, 0);
+		}
+	}
+
+	_onChange(e) {
+		if (e.target.getAttribute('role') !== 'menuitemradio') {
+			return;
+		}
+		this.close();
+	}
+
+	_onSelect(e) {
+		if (e.target.tagName !== 'D2L-MENU-ITEM') {
+			return;
+		}
+		this.close();
+	}
+
+	__getMenuElement() {
+		return this.shadowRoot.querySelector('slot')
+			.assignedNodes()

```

Comment:
Curious if `assignedElements()` works in all browsers now?  I think Safari might have been the hold-out... but based on a quick check it seems like it works for me in one of my StackBlitz projects.  It may avoid the check for `hasAttribute`. If it works, there are other places that are using `assignedNodes()` and explicitly filtering to get elements - they could eventually be cleaned up if this API is fully supported now.  Not a big deal though.

Reply:
Technically we still support Safari 11, so we'll need to be careful.